src/main/java/com/iota/iri/utils/log/interval/IntervalLogger.java

```diff
@@ -0,0 +1,490 @@
+     * is the same as dumping the message manually through the {@link org.slf4j.Logger} object itself.
+     */
+    private void scheduleOutput() {
+        if (outputThread == null) {
+            synchronized(this) {
+                if (outputThread == null) {
+                    outputThread = new Thread(() -> {
+                        try {
+                            Thread.sleep(Math.max(logInterval - (System.currentTimeMillis() - lastLogTime), 1));
+
+                            outputThread = null;
+
+                            triggerOutput(true);
+                        } catch(InterruptedException e) {
+                            outputThread = null;
+
+                            Thread.currentThread().interrupt();
+                        }
+                    }, Thread.currentThread().getName());
+
+                    outputThread.start();
+                }
+            }
+        }
+    }
+
+    /**
+     * Represents the basic logic that is shared by all message types that are supported by this logger.
+     */
+    private abstract class Message {
+        /**
+         * Holds the message that shall get printed.
+         */
+        protected final String message;
+
+        /**
+         * Creates a message that gets managed by this logger.
+         *
+         * It simply stores the provided message in the internal property.
+         *
+         * @param message message that shall get printed
+         */
+        public Message(String message) {
+            this.message = message;
+        }
+
+        /**
+         * This method handles the actual output of messages through the {@link org.slf4j.Logger} instance.
+         *
+         * It first checks if the message that shall get printed differs from the last message that was printed and then
+         * issues the output of the message. After printing the message, it updates the internal variables to handle the
+         * next message accordingly.
+         */
+        public abstract void print();
+
+        /**
+         * This method allows us to used the Messages in HashMaps and other collections that use the hashCode method to
+         * identify objects. Even tho this is not actively used we anyway want to implement it to be on the safe side
+         * if we ever implement something like this.
+         *
+         * @return a unique int identifier that can be used to address equal objects
+         */
+        @Override
+        public abstract int hashCode();
+
+        /**
+         * This method allows us to compare different {@link Message}s for equality and check if we have received or
+         * processed the same {@link Message} already.
+         *
+         * @param obj object that shall be compared to this instance
+         * @return true if it represents the same message or false otherwise
+         */
+        @Override
+        public abstract boolean equals(Object obj);
```

Comment:
But thats not sufficient to detect the equality of objects? It would simply allow me to replace the 

            if (!(obj instanceof InfoMessage)) {
                return false;
            }

by

            if (!super.equals(obj)) {
                return false;
            }

which isnt much better?!?

Reply:
In all classes you just compare message.
In `ErrorMessage` you can do:
`super.equals(obj) && Objects.equals(cause, (ErrorMessage) obj).cause)`

I just noticed you wrote `cause == ((ErrorMessage) obj).cause` which is a bug imo.